app/main.py

```diff
@@ -1,3 +1,67 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: (Distance | int | float)) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        elif isinstance(other, float) or isinstance(other, int):
+            return Distance(self.km + other)
+        else:
+            return self
+
+    def __iadd__(self, other: (Distance | int | float)) -> Distance:
+        if isinstance(other, Distance):
+            self.km = self.km + other.km
+        if isinstance(other, float) or isinstance(other, int):
+            self.km = self.km + other
+        return self
+
+    def __mul__(self, num: float) -> Distance:
```

Comment:
Indicate that num could be `int` as well here and in the other places in the type annotation